tests/CMakeLists.txt

```diff
@@ -1,15 +1,22 @@
 # Require at least version 2.8.12 of CMake
 cmake_minimum_required(VERSION 2.8.12)
 
-# Include global configure settings
-include(${CMAKE_CURRENT_SOURCE_DIR}/../configure.cmake)
-include(${CMAKE_CURRENT_SOURCE_DIR}/../configurecompiler.cmake)
+if (NOT WIN32)
+  set(CMAKE_CXX_FLAGS_INIT "${CMAKE_CXX_FLAGS_INIT} -std=c++11"  CACHE STRING "Flags used by the compiler during all build types.")
+  set(CMAKE_C_FLAGS_INIT "${CMAKE_C_FLAGS_INIT} -std=c11"  CACHE STRING "Flags used by the compiler during all build types.")
```

Comment:
The only C11 feature used in CoreCLR is `_Static_assert` in `src/corefx/System.Globalization.Native/pal_compiler.h`. The rest of libunwind and Globalization is gnu99 compatible (c99 plus GNU extensions: named variadic macros and statement expressions etc.). However, `-pedantic`ally, I think we need `-std=gnu11 -pedantic -Wno-gnu -Wno-variadic-macros -Wno-format-pedantic -Wno-pointer-arith -Wno-keyword-macro` for C in CoreCLR.

CoreFX uses gnu99 (the initial reason was: mono compatibility) but the code has C11's `_Generic` as well as `_Static_assert`, which makes it non-conforming to the compiler's `-std` filag. IMO, we should select the minimal standard that is actually used by the code (`-pedantic` highlights where we violate the conformance).

Reply:
The `_Static_assert` is optional in both coreclr and corefx (we check if it exists and if not, we use another way). As for the `_Generic` in corefx, you are right.
I would not change this stuff in this change though.

Reply:
Sure, btw I have this fixed in ongoing solaris port branch, I will rebase once this PR is merged.
gcc 7 on SmartOS amd64 throws the following error on building Globalization target with `-std=c11`, with gets fixed with `-std=gnu99` or `-std=gnu11` flag (without changing the code or other flags):

```sh
/datadrive/projects/coreclr/src/corefx/System.Globalization.Native/pal_calendarData.c:84:9: error: implicit declaration of function 'strcasecmp'; did you mean 'u_strcasecmp'? [-Werror=implicit-function-declaration]
     if (strcasecmp(calendarName, GREGORIAN_NAME) == 0)
         ^~~~~~~~~~
         u_strcasecmp
```

Reply:
That warning I would think is actually a problem you should look at fixing (the declaration of `strcasecmp` isn't found anywhere) even though the "implicit definition" behavior happens to match the signature.